examples/01_dirty_categories.py

```diff
@@ -1,398 +1,154 @@
-# as low cardinality string variables.
-# A |OneHotEncoder| will be applied to these columns.
-#
-# The vectorizer actually makes the difference between string variables
-# (data type ``object`` and ``string``) and categorical variables
-# (data type ``category``).
-#
-# Next, we can have a look at the encoded feature names.
+# What just happened there?
+# -------------------------
 #
-# Before encoding:
-X.columns.to_list()
+# First, it did not raise any errors, yay!
+# Let's explore the internals of our encoder, the |TableVectorizer|:
 
-###############################################################################
-# After encoding (we only plot the first 8 feature names):
-feature_names = table_vec.get_feature_names_out()
-feature_names[:8]
+from pprint import pprint
 
-###############################################################################
-# As we can see, it gave us interpretable columns.
-# This is because we used the |Gap| on the column 'division',
-# which was classified as a high cardinality string variable.
-# (default values, see |TV|'s docstring).
-#
-# In total, we have a reasonable number of encoded columns:
-len(feature_names)
+# Recover the TableVectorizer from the pipeline
+tv = pipeline.named_steps["tablevectorizer"]
 
+pprint(tv.transformers_)
 
 ###############################################################################
-# Feature importances in the statistical model
-# --------------------------------------------
-#
-# In this section, we will train a regressor, and plot the feature importances.
+# We observe it has automatically assigned an appropriate encoder to
+# corresponding columns.
+# For example, it classified the columns 'gender', 'department',
+# 'department_name' and 'assignment_category' as low cardinality
+# string variables.
+# Two remarkable things:, it has affected a |GapEncoder| to the columns
+# `employee_position_title` and `division`, and a |DatetimeEncoder| to the
+# 'date_first_hired' column.
 #
-# .. topic:: Note:
+# The |GapEncoder| is a powerful encoder that can handle dirty
+# categorical columns.
+# The |DatetimeEncoder| can encode datetime columns for machine learning.
 #
-#    To minimize computation time, we use the feature importances computed by the
-#    |RandomForestRegressor|, but you should prefer |permutation importances|
-#    instead (which are less subject to biases).
+# We will explore those in the next sections.
 #
-# First, let's train the |RandomForestRegressor|:
-
-from sklearn.ensemble import RandomForestRegressor
-
-regressor = RandomForestRegressor()
-regressor.fit(X_train_enc, y_train)
-
-###############################################################################
-# Retrieving the feature importances:
 
-importances = regressor.feature_importances_
-std = np.std([tree.feature_importances_ for tree in regressor.estimators_], axis=0)
-indices = np.argsort(importances)
-# Sort from least to most
-indices = list(reversed(indices))
+# TODO
```

Comment:
It seems to me that there is still a bit of work.

At least, we should put back the discussion on the feature names from the original example.